stdlib/mexpr/eq.mc

```diff
@@ -97,7 +100,12 @@ lang Eq
   sem eqExprH (env : EqEnv) (free : EqEnv) (lhs : Expr) =
   -- Intentionally left blank
 
-  sem eqType (typeEnv : EqTypeEnv) (lhs : Type) =
+  sem eqType (lhs : Type) =
```

Comment:
Would they? Currently free variables are checked for symbol equality in `eqType`, so the `a` and the `b` would be different, wouldn't they?

As an aside, there doesn't actually seem to be a case in `eqExpr` for `TmType` currently.

Reply:
Aren't they assumed to be equal when they're found (and both unbound) so that the equality works modulo alpha-conversion (i.e. `lam x. y` = `lam z. w`, we assume `y = w` when we find them unbound and see if we later find an inconsistent assumption)? I believe that's the case for `TmVar` (that's why it returns an `Option` and not just a `Bool`) and figured it was the same idea for free `TyVar`, in which case the "these are equal" assumption must carry through so we don't get inconsistent assumptions.

Reply:
Right, I didn't implement that yet because of laziness and not wanting to add too much stuff to the PR, but I probably should.